protocols/pop3/src/main/java/org/apache/james/protocols/pop3/core/POP3MessageCommandArguments.java

```diff
@@ -0,0 +1,81 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.protocols.pop3.core;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.apache.james.protocols.api.Request;
+
+public class POP3MessageCommandArguments {
+    
+    private static POP3MessageCommandArguments INVALID = new POP3MessageCommandArguments(false, 0, Optional.empty());
+    
+    private final boolean valid;
+    private final int messageNumber;
+    private final Optional<Integer> lineCount;
+
+    public static POP3MessageCommandArguments fromRequest(Request request) {
```

Comment:
```suggestion
    public static Optional<POP3MessageCommandArguments> fromRequest(Request request) {
```

`private final boolean valid` is clearly an anti-pattern as we purpously build some invalid objects. Let's just avoid return them in the first place!

(actually the magic constants ture/false and 0 catched my eyes, I was about to comment about them but then looking deeper it seems there's more.)

Reply:
I originally thought about Optional, but it is a bit awkward here since I would need something like .orElseReturn(...) which obviously does not exist.
Anyway, it is safer to use with Optional, so I changed it in the latest push. Bonus, no more magic numbers/constants.